kafka-client/src/main/java/dev/responsive/kafka/internal/db/RemoteKVTable.java

```diff
@@ -19,20 +19,20 @@
 import org.apache.kafka.common.utils.Bytes;
 import org.apache.kafka.streams.state.KeyValueIterator;
 
-public interface RemoteKVTable<S> extends RemoteTable<Bytes, S> {
+public interface RemoteKVTable<P, S> extends RemoteTable<Bytes, P, S> {
```

Comment:
Was able to remove the `P` generic from the `RemoteTable` by just using a `?` type for the WriterFactory returned by `RemoteTable#init`. And because the partitioner was extracted out of the `KeyValue/WindowOperations` class, we actually don't have to care about the WriterFactory type and just pass it directly through to the CommitBuffer. So we don't even have to do any casting after all